### DISCUSSION TITLE
Miten saan enchant-2 komennon huomioimaan kaikki sanat, jotka löytyy Joukahaisesta?

### DISCUSSION THREAD

**Discussion by ylivuoto** [voikko/discussion#23] [Q&A] [answered]
Terve!

Asensin editoriini wcheck-moden ja sitä varten libenchant-2-voikko paketin. Tällä setupilla enchant-2 kuitenkin valittaa mm. sanasta "iteratiivinen". Sana kuitenkin löytyy hakemalla Joukahaisesta, [https://joukahainen.puimula.org/word/edit?wid=525836](https://joukahainen.puimula.org/word/edit?wid=525836).

Samoin [https://oikofix.com](https://oikofix.com) valittaa sanasta "iteratiivinen". Miten saan laajennettua enchant-2 toimimaan vaikkapa kaikilla Joukahaiseen lisätyillä sanoilla?

T.,
O-P

Edit: lisäsin käyttöjärjestelmän tiedot
uname -a (nodename kumitettu):

--kernel-name Linux
--kernel-release 6.8.0-40-generic
--kernel-version #40~22.04.3-Ubuntu SMP PREEMPT_DYNAMIC Tue Jul 30 17:30:19 UTC 2 
--machine x86_64 
--processor x86_64 
--hardware-platform x86_64 
--operating-system GNU/Linux

**Comment by ahomansikka** [comment, marked as answer]
Iteratiivisen käyttöalaksi on laitettu it eli tietojenkäsittely
`<usage><flag>it</flag><usage>`
ja standardi-Voikko ei ota oikolukuun mukaan tällaisia sanoja.

Eräs ratkaisu on kääntää Voikko niin, että tietojenkäsittelysanat tulevat mukaan.
Vihjeitä käätämiseen voi katsoa Makefile'n kohdasta vvfst-sukija:

**Reply by tlikonen** [reply]
Jos haluaa kaikki Joukahaisen sanat, täytyy todellakin kääntää Voikko siten, että erityisalojen sanastot tulevat mukaan.

**Reply by ylivuoto** [reply]
Kiitos! Miten pystyn komentoriviltä testaamaan, onnistuiko toi käännös? Asensin Voikon siis aiemmin valmiista paketeista, enkä ole täysin varma menikö sekään asennus ihan kunnolla purkkiin.

**Comment by tlikonen** [comment]
”Iteratiivinen” on kielitoimiston sanakirjassakin, eikä sillä ole siellä merkintää, että kuuluisi erityisalan sanastoon. Poistin atk-lipun Joukaisen tietueesta, joten seuraavissa versioissa ”iteratiivinen” on mukana perussanastossa. https://joukahainen.puimula.org/word/edit?wid=525836

**Comment by ahomansikka** [comment]
Linuxissa:

grep '<form>' joukahainen.xml | gawk '{print substr($0,9)}'|gawk -F'<' '{print $1}' |sed -e 's/=//g'|voikkospell

Kaikki yhdelle riville.
Tulostuksen rivit, jotka alkavat W:
ovat sanoja, jotka eivät kuulu oikolukusanastoon.

pe, 2024-10-25 kello 04:04 -0700, O-P kirjoitti:

**Comment by ahomansikka** [comment]
Tarkennus:

W:-alkuiset kuuluvat sanastoon, jota sinun Voikko-versiosi ei tunnista.
Jos kaikki meni hyvin, ne ovat vain sanoja, jotka eivät kuulu
oikolukusanastoon.

Jos haluat testata, tunnistaako Voikko-versiosi kaikki Joukahaisen it-sanat, sinun pitää poimia ne tiedostosta joukahainen.xml ja testata samalla tavalla.

Pitää etsiä sanat, joissa on tämmöistä...

	<usage>
		<flag>it</flag>
	</usage>

Jokohan nyt meni oikein...

pe, 2024-10-25 kello 04:04 -0700, O-P kirjoitti:
